ProfilerGUI/Source/Configurator/UploadViewModel.cs

```diff
@@ -0,0 +1,193 @@
+{
+    /// <summary>
+    /// View model for configuring the uploader daemon.
+    /// </summary>
+    internal class UploadViewModel : INotifyPropertyChanged
+    {
+        private static readonly Logger logger = LogManager.GetCurrentClassLogger();
+
+        private readonly HttpClient client = new HttpClient();
+
+        /// <summary>
+        /// <inheritDoc />
+        /// </summary>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// The config for the daemon.
+        /// </summary>
+        public UploadConfig Config { get; private set; } = null;
+
+        /// <summary>
+        /// Whether the configuration UI for the Teamscale server should be shown.
+        ///
+        /// The three cases are modeled as follows:
+        /// - no upload: Config == null
+        /// - upload to Teamscale: Config != null && Config.Teamscale != null
+        /// - upload to directory: Config != null && Config.Teamcsale == null
+        /// </summary>
+        public int UploadTypeIndex
+        {
+            get
+            {
+                if (Config == null)
+                {
+                    return 0;
+                }
+                if (Config.Teamscale != null)
+                {
+                    return 1;
+                }
+                return 2;
+            }
+            set
+            {
+                switch (value)
+                {
+                    case 0:
+                        Config = null;
+                        break;
+
+                    case 1:
+                        if (Config == null)
+                        {
+                            Config = new UploadConfig();
+                        }
+                        Config.Teamscale = new TeamscaleServer();
+                        Config.Directory = null;
+                        break;
+
+                    case 2:
+                        if (Config == null)
+                        {
+                            Config = new UploadConfig();
+                        }
+                        Config.Teamscale = null;
+                        Config.Directory = string.Empty;
+                        break;
+                }
+
+                // force all UI elements to be refreshed since this affects almost all of them
+                PropertyChanged.Raise(this, null);
+            }
+        }
+
+        /// <summary>
+        /// The directory to upload traces to.
+        /// </summary>
+        public string Directory
+        {
+            get => Config?.Directory;
+            set
+            {
+                Config.Directory = value;
+                PropertyChanged.Raise(this);
+            }
+        }
+
+        /// <summary>
+        /// Whether the configuration UI for the Teamscale server should be shown.
+        /// </summary>
+        public bool IsTeamscaleConfigVisible { get => Config != null && Config.Teamscale != null; }
```

Comment:
IsTeamscaleConfigVisible => Config != ...
no need to use curlies and get :)

Reply:
didn't know that! :-) done